addons/hr_holidays/models/hr_leave.py

```diff
@@ -881,6 +866,18 @@ def create(self, vals_list):
                 # Handle double validation
                 if mapped_validation_type[leave_type_id] == 'both':
                     self._check_double_validation_rules(employee_id, values.get('state', False))
+                if values.get('date_from') and values.get('date_to'):
+                    if employee_id:
+                        employee_resource = self.env['hr.employee'].browse(employee_id).resource_id
+                        employee_tz = timezone(employee_resource.tz)
+                        leave_date_from, leave_date_to = None, None
+                        # Set date_from and date_to in the employee's timezone if date_from and date_to are datetime objects
+                        leave_date_from = get_datetime_from_string(values.get('date_from')).replace(tzinfo=employee_tz)
+                        leave_date_to = get_datetime_from_string(values.get('date_to')).replace(tzinfo=employee_tz)
+                        # Search for the employee start and end of the day according to his/her calendar
+                        leave_date_from, leave_date_to = employee_resource._adjust_to_calendar(leave_date_from, leave_date_to)[employee_resource]
+                        values['date_from'] = (leave_date_from and leave_date_from.astimezone(pytz.utc).replace(tzinfo=None)) or values['date_from']
+                        values['date_to'] = (leave_date_to and leave_date_to.astimezone(pytz.utc).replace(tzinfo=None)) or values['date_to']
```

Comment:
I think this whole logic should be done in `_compute_date_from_to` instead, what do you think?